runtime/sema/entitlementset.go

```diff
@@ -141,14 +153,38 @@ func (s *EntitlementSet) Minimize() {
 	}
 }
 
+// Returns whether this entitlement set is minimally representable in Cadence.
+//
+// If true, this set can be exactly represented as a non-nested logical formula: i.e. either a single conjunction or a single disjunction
+// If false, this set cannot be represented without nesting connective operators, and thus must be over-approximated when being
+// represented in Cadence.
+// As Cadence does not support nesting disjunctions and conjunctions in the same entitlement set, this function returns false
+// when s.Entitlements and s.Disjunctions are both non-empty, or when s.Disjunctions has more than one element
+func (s *EntitlementSet) IsMinimallyRepresentable() bool {
+	if s == nil {
+		return true
+	}
+
+	if !s.minimized {
+		s.Minimize()
+	}
+
+	return s.Disjunctions.Len() == 0 || (s.Entitlements.Len() == 0 && s.Disjunctions.Len() == 1)
```

Comment:
Isn't this minimally representable: `(s.Entitlements.Len() > 0 && s.Disjunctions.Len() == 0)`?
Or is it not possible to have this state?